src/Microsoft.Health.Dicom.Core/Features/Store/StoreDatasetValidator.cs

```diff
@@ -202,14 +205,16 @@ string EnsureRequiredTagIsPresent(DicomTag dicomTag)
                     {
                         validationResultBuilder.Add(ex, item.Tag);
                     }
+
+                    bool isIndexableTag = indexableTags.Contains(item.Tag);
```

Comment:
Instead of creating a list from above, you can search `queryTags` directly like: `bool isIndexableTag = queryTags.Any(x => x.Tag == item.Tag)`. For larger collections, I'd say to move it into a set, but I think this thing can only be 126 for extended query tags + any core ones.